homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,232 @@
+        if user_input is None or errors is not None:
+            return self.async_show_form(
+                step_id="add_account", data_schema=ACCOUNT_SCHEMA, errors=errors
+            )
+        return await self.async_handle_data_and_route(user_input)
+
+    async def async_handle_data_and_route(self, user_input: ConfigType):
+        """Handle the user_input, check if configured and route to the right next step or create entry."""
+        self._update_data(user_input)
+        if self._data and self._port_already_configured():
+            return self.async_abort(reason="already_configured")
+
+        if user_input[CONF_ADDITIONAL_ACCOUNTS]:
+            return await self.async_step_add_account()
+        return self.async_create_entry(
+            title=TITLE.format(self._data[CONF_PORT]),
+            data=self._data,
+            options=self._options,
+        )
+
+    def _update_data(self, user_input: ConfigType) -> None:
+        """Parse the user_input and store in data and options attributes.
+
+        If there is a port in the input or no data, assume it is fully new and overwrite.
+        Add the default options and overwrite the zones in options.
+        """
+        if not self._data or user_input.get(CONF_PORT):
+            self._data = {
+                CONF_PORT: user_input[CONF_PORT],
+                CONF_PROTOCOL: user_input[CONF_PROTOCOL],
+                CONF_ACCOUNTS: [],
+            }
+        account = user_input[CONF_ACCOUNT]
+        self._data[CONF_ACCOUNTS].append(
+            {
+                CONF_ACCOUNT: account,
+                CONF_ENCRYPTION_KEY: user_input.get(CONF_ENCRYPTION_KEY),
+                CONF_PING_INTERVAL: user_input[CONF_PING_INTERVAL],
+            }
+        )
+        self._options[CONF_ACCOUNTS].setdefault(account, deepcopy(DEFAULT_OPTIONS))
+        self._options[CONF_ACCOUNTS][account][CONF_ZONES] = user_input[CONF_ZONES]
+
+    def _port_already_configured(self):
+        """See if we already have a SIA entry matching the port."""
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.data[CONF_PORT] == self._data[CONF_PORT]:
+                return True
+        return False
+
+
+class SIAOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle SIA options."""
+
+    def __init__(self, config_entry):
+        """Initialize SIA options flow."""
+        self.config_entry = config_entry
+        self.options = deepcopy(dict(config_entry.options))
+        self.hub: SIAHub | None = None
+        self.accounts_todo: list = []
+
+    async def async_step_init(self, user_input: ConfigType = None):
+        """Manage the SIA options."""
+        self.hub = self.hass.data[DOMAIN][self.config_entry.entry_id]
+        if self.hub is not None and self.hub.sia_accounts is not None:
+            self.accounts_todo = [a.account_id for a in self.hub.sia_accounts]
+            return await self.async_step_options()
```

Comment:
We have an inconsistent return value here. Instead we should abort the flow if the hub isn't set up yet.

Reply:
you should only be able to run this code when the hub is created anyway, but type checking requires I check if the hub variable is not None, and thus we get this, I also checked and I didn't see default abort reasons that would match this case (because this case should not be possible), so let me know how to handle this!

Reply:
If it's just for typing reasons we can assert the hub instead.

We can always define new abort reasons. That's not a problem.

Reply:
so add `assert self.hub is not None` at 187?

Reply:
Yes, sounds correct.